### PR TITLE
4116 catalogue items list is querying twice for each render

### LINKED ISSUE #4116: Catalogue Items List is querying twice for each render
## What went wrong? 😲

Catalogue Items list queries twice on each render

Only occurs if user first clicks into detail view, and then clicks back to list view.

Maybe a global issue?

https://github.com/msupply-foundation/open-msupply/assets/90807420/e63ed903-9830-4394-a383-3540c3a8c73b

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #4116

# 👩🏻‍💻 What does this PR do?
Fix itemWithStats being called twice 

# 🧪 Testing

- [ ] Go to item page and check network to see if api still being called twice

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/system/src/Item/ListView/ListView.tsx [hunk 1]
```diff
@@ -108,7 +111,12 @@ const ItemListComponent: FC = () => {
         isError={isError}
         isLoading={isLoading}
         onRowClick={row => {
-          navigate(`/catalogue/items/${row.id}`);
+          navigate(
```
Comment (roxy-dao) [inline client/packages/system/src/Item/ListView/ListView.tsx:114]: Sideqwest~

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Oops, this broke the items filter. 😢

Screenshot 2024-08-23 at 12 52 39 PM

I'll try and fix it this afternoon.

**Comment by andreievg** [comment]
I see that Laché assigned @CarlosNZ to fix, which leaves someone else to assign as reviewer, thank you

**Comment by CarlosNZ** [comment]
@andreievg, have had a look at this.  Was planning to pause on it until I can inspect what's happening on Production server -- I suspect it may not be needed at all.

**Comment by CarlosNZ** [comment]
@lache-melvin @andreievg 

So, it seems that this issue occurs on most of the List and Detail view components that are using the original API pattern. In the places where we've refactored it (Stock, Reports), it's behaving as it should (i.e. single GraphQL query on load). 

So I don't think we need to fix this specific case. The problem will naturally go away as we gradually refactor the API hooks to the new pattern.

Okay to close this?

**Comment by lache-melvin** [comment]
Thanks for investigating Carl! Yeah that sounds good to me, another motivator to keep refactoring to the new pattern! 😁

**Comment by CarlosNZ** [comment]
Closing as per [above comment](https://github.com/msupply-foundation/open-msupply/pull/4669#issuecomment-2319718938)

**Pull request closed without merging**